### ISSUE TITLE
Inducing GP prior varies by location, but support contraction uses one prior SD

### ISSUE THREAD

**Issue by bschilder** [bschilder/genomeOS#266]
## Scientific problem

The data-support mask interprets posterior SD / prior SD as learning from data
(design §7.1b). The inducing implementation has a location-dependent approximate
prior, but the denominator is sampled at the first observation coordinate and
reused everywhere. A difference between locations' priors can consequently cross
the contraction threshold even when the local distribution has not changed.

This advances #189 WP2 and #39. It is separate from #129's question of whether
biological correlation lengths should vary spatially. This report proposes no
change to the 0.9 threshold, the engine decision #34, or published artifacts.

## Source trace

Main was verified at `3c986902c12b40b84c56145ed863f7f519cbab6b` on September11.

- [`SurfaceFit.prior_frequency_sd`](https://github.com/bschilder/genomeOS/blob/3c986902c12b40b84c56145ed863f7f519cbab6b/genomeos/surfaces/fit.py#L350)
  explicitly justifies one scalar by prior stationarity.
- [`x_pred` starts at `x[:1]`](https://github.com/bschilder/genomeOS/blob/3c986902c12b40b84c56145ed863f7f519cbab6b/genomeos/surfaces/fit.py#L712).
- The [inducing training and prediction expressions](https://github.com/bschilder/genomeOS/blob/3c986902c12b40b84c56145ed863f7f519cbab6b/genomeos/surfaces/fit.py#L743)
  are deterministic given inducing coefficients. Their field covariance is
  `Q = K_xu (K_uu + jitter I)^-1 K_ux`, whose diagonal depends on the query location.
- The [scalar is estimated from that first-location prior](https://github.com/bschilder/genomeOS/blob/3c986902c12b40b84c56145ed863f7f519cbab6b/genomeos/surfaces/fit.py#L852)
  and [used for all cells](https://github.com/bschilder/genomeOS/blob/3c986902c12b40b84c56145ed863f7f519cbab6b/genomeos/surfaces/mask.py#L168).

This is the effective-prior distinction described by
[Quinonero-Candela and Rasmussen (2005), §§2,4](https://www.jmlr.org/papers/volume6/quinonero-candela05a/quinonero-candela05a.pdf).
The stationary parent kernel does not make its deterministic finite-rank
projection stationary. The omitted covariance `K-Q` is an additional approximation
question; correcting a denominator alone does not restore it.

## Executed evidence and its limits

A frozen synthetic diagnostic evaluated both public H3/k-means placements,
budgets16/32/48, lengths300/750/1500/3000km:24cases,81queries each. All264 checks
passed: actual PyMC kernel parity, independent covariance solves, PSD/bounds,
zero/jitter inducing-site controls,128/256-point logistic-normal quadrature and
three independently integrated query moments per case. Maximum kernel difference
was9.13e-13; independent-solve difference1.23e-15; adaptive-moment difference5.56e-16.
Execution exited0 with empty stderr. Environment: Python3.12.13, NumPy2.4.6,
SciPy1.18.1, PyMC6.3.1, PyTensor3.3.0, H3 4.5.0.

The diagnostic fixes amplitude1 and each declared length, retaining the production
Normal(-3.5,1.5) intercept. **It conditions on hyperparameters; it does not integrate
the production amplitude/lengthscale priors or fit counts.** Identical local prior
and hypothetical posterior distributions isolate the normalization behavior.
No real source, posterior, published mask or current global-accuracy claim is tested.
All results are `automated_proposal/pending`; publication eligibility is false.

Two of24configurations produced classification differences. These illustrative
counterexamples were selected after inspecting the complete matrix:

| Configuration | In-range queries | `interpolated` with first-location denominator | `interpolated` with matched local denominator |
|---|---:|---:|---:|
| H3,16inducing points,1500km |77|2|0|
| k-means,48inducing points,750km |77|6|0|

The other22configurations had no state disagreement. The four far-control points
were retained as `unknown`; no failed or inconvenient case was removed.

## Minimal reproduction

Run in the repository's locked Atlas environment, with its source on PYTHONPATH.
This is authored synthetic geometry, not population or recruitment metadata.

```python
from itertools import product

import numpy as np
from scipy.linalg import solve_triangular
from scipy.special import expit, roots_hermitenorm

from genomeos.surfaces.fit import EARTH_RADIUS_KM, JITTER, h3_inducing_points, to_unit_sphere
from genomeos.surfaces.mask import MaskConfig, classify_support

anchors = np.array(list(product(range(-30, 41, 10), range(-20, 69, 8))), dtype=float)
u = h3_inducing_points(anchors[:, 0], anchors[:, 1], 16, 1500.0)
x = to_unit_sphere([anchors[0, 0], -25.0], [anchors[0, 1], 64.0])
length = 1500.0 / EARTH_RADIUS_KM

def kernel(a, b):
    r = np.sqrt(5.0) * np.linalg.norm(a[:, None, :] - b[None, :, :], axis=2) / length
    return (1.0 + r + r*r/3.0) * np.exp(-r)

chol = np.linalg.cholesky(kernel(u, u) + JITTER*np.eye(len(u)))
basis = solve_triangular(chol, kernel(x, u).T, lower=True).T
variance = 1.5**2 + np.sum(basis*basis, axis=1)
nodes, weights = roots_hermitenorm(256)
weights = weights / np.sqrt(2*np.pi)
p = expit(-3.5 + np.sqrt(variance[:, None])*nodes)
sd = np.sqrt(p*p @ weights - (p @ weights)**2)
anchor_xyz = to_unit_sphere(anchors[:, 0], anchors[:, 1])
chord = np.linalg.norm(x[1] - anchor_xyz, axis=1).min()
distance = 2*EARTH_RADIUS_KM*np.arcsin(chord/2)
for denominator in (sd[0], sd[1]):
    ratio = sd[1] / denominator
    state = classify_support(
        has_observation_centre=np.array([False]),
        dist_nearest_obs_km=np.array([distance]),
        posterior_contraction=np.array([ratio]),
        correlation_range_km=1500.0,
        config=MaskConfig(),
    )[0]
    print(float(ratio), state)
```

Observed: approximately `0.8953380210217113 interpolated`, then
`1.0 prior_dominated`. The distribution at the query is unchanged in both calls;
only the prior denominator changes.

## Acceptance evidence for a repair

- Specify whether contraction is measured against the implemented approximate
  prior or the intended parent GP, and make numerator/denominator consistent.
- Add a coordinate-aware prior calculation or another justified correction;
  do not label a single-location statistic stationary without demonstrating it.
- Test unchanged-prior normalization, query ordering/batching, H3/k-means coverage,
  HSGP separately, and save/load reproducibility on the actual prior-prediction path.
- Quantify the separate `K-Q` approximation using a small full-GP/count reference
  before claiming calibrated fitted uncertainty. No full inference or threshold
  change is authorized by the conditional diagnostic alone.
- Preserve existing cited artifacts; any corrected mask/model gets a new version.

Expert review should check the normalization target, conditional-to-full-prior
scope, approximation uncertainty, and implications for §7.1b support classification.

**Comment by bschilder** [comment]
### WP2 follow-up: a tractable full-GP/count reference

The separate covariance question now has a bounded count-conditioned diagnostic.
This is authored synthetic evidence, `automated_proposal/pending`, with no real
fit, artifact publication, threshold change or global-accuracy claim.

The frozen matrix contains 24 inducing fields (H3/k-means, budgets16/32/48,
lengths300/750/1500/3000km) and four matching full fields, each evaluated at81
queries. Amplitude1, intercept Normal(-3.5,1.5), jitter1e-4 and sparse geometry
are fixed. Exactly one site has a likelihood: no-update control or counts0/20,
2/20,10/20 under binomial and concentration20 beta-binomial laws. The96 geometry
anchors are not96 observations. Both fields include the same shared intercept;
the full field uses K and the sparse field uses Q.

All196 cases and1,194 numerical checks passed. Latent frequency moments were
computed by Gaussian conditioning, Gauss-Hermite inner integration and adaptive
outer integration. Twenty-seven separately evaluated nested-adaptive scalar
references agreed to maximum1.73e-13 in means and2.65e-13 in SDs. The count-law
reference used direct small-count products. No MCMC or inferred hyperparameter
estimate enters these comparisons.

**The approximation effect survives correct local normalization.** An example
chosen from the completed matrix uses H3 budget16, length1500km, a2/20
beta-binomial observation at(-30,-20), and query(-25,8),2809.664km away:

| Quantity | Sparse field | Full field |
|---|---:|---:|
| Posterior SD / own local prior SD |0.754215266|0.918921547|

Sparse/full posterior SD is0.655027067. Both ratios are internally consistent,
yet lie on opposite sides of the unchanged0.9 criterion. Across the complete
in-range matrix, opposite-side comparisons occur in6/720 binomial and42/720
beta-binomial query/configuration/outcome combinations. These overlapping
synthetic combinations are not independent observations or a real-map prevalence
estimate. All four far controls remain outside twice the fixed range.

This supports proceeding with the denominator repair while keeping covariance
approximation validation separate. The calculation does not integrate production
hyperpriors, fit multiple observations, include HSGP, or establish calibrated
real-world uncertainty. The full GP is a mathematical reference, not biological
ground truth. No-update ratios agreed with1 to numerical precision.

The complete matrix, quadrature/error records, frozen plan and executed source
are retained. Plan SHA256:
`44dcbbaebeb8efeb2e565075e99d73470c9dad424238ec30b4758d182cdafb19`.
Executed script SHA256:
`6d3156c11e395aabfea2b8cc8174e21262e8d9092c5f120123b3afde4caece1f`.
Both were rehashed after execution and matched the result record. Execution
exited0 with empty stderr;40 smoke checks passed. This advances #189 WP2.
